app_dart/test/src/service/fake_firestore_service.dart

```diff
@@ -502,11 +502,20 @@ abstract base class _FakeInMemoryFirestoreService
       return collectionId == collection;
     });
 
-    results = results.where((document) {
-      return _matchesFilter(document.fields!, filterMap);
-    });
-
-    if (compositeFilterOp != kCompositeFilterOpAnd) {
+    if (compositeFilterOp == kCompositeFilterOpAnd) {
+      results = results.where((document) {
+        return _matchesFilter(document.fields!, filterMap);
+      });
+    } else if (compositeFilterOp == kCompositeFilterOpOr) {
+      final listFilters = [
```

Comment:
I'm going to ignore the part where Firebase allows for nested filters, so you could do nested and/or.